### PR TITLE
feat: Adds Kind extension

### LINKED ISSUE #413: Kind support
### Epic domain

Podman Desktop should provide the ability to enable and start [Kind](https://kind.sigs.k8s.io/) as an available provider.

Podman provides the ability to run Pods which helps developers to reproduce workloads that are running on kubernetes in a local environment.

In addition, as Podman Desktop has the ability to handle multiple providers (container engines or kubernetes distribution) it would be nice to allow users to bootstrap a "local kubernetes" with Kind. (We saw asks from the community on this regards too.)

### Sub-tasks

- Ability to setup Kind
- Ability to start/stop Kind
- Display Pods from Kind in the list of Pods

### Additional context

Additionally, we would like to enable a set of "Developer packages" that the developer could eventually enable. As mentioned by @cmoulliard in his [comment](https://github.com/containers/podman-desktop/issues/29#issuecomment-1118231429).

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Allow to call kind tool and
- list kind clusters
- Allows to create clusters

draft as it requires all my PR being merged first

### Screenshot/screencast of this PR

https://user-images.githubusercontent.com/436777/208451335-fc134ad7-21fb-4011-a632-cc681d5787ee.mp4

### What issues does this PR fix or reference?

Fixes #413 

### How to test this PR?

Go to settings/resources/Kind menu
- if you have some kind clusters you should see them
- you should be able to create one cluster

Change-Id: Ib9174b6a453d1970fe4bec54fb29013ebd5db436
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: extensions/kind/package.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            "docker"
+          ],
+          "scope": "KubernetesProviderConnectionFactory",
+          "description": "Provider Type"
+        },
+        "kind.cluster.creation.http.port": {
+          "type": "number",
+          "default": 9090,
+          "scope": "KubernetesProviderConnectionFactory",
+          "description": "HTTP port"
```
Comment (cdrage) [inline extensions/kind/package.json]:
```suggestion
          "description": "HTTP Port"
```

File: extensions/kind/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        },
+        "kind.cluster.creation.https.port": {
+          "type": "number",
+          "default": 9443,
+          "scope": "KubernetesProviderConnectionFactory",
+          "description": "HTTPS port"
```
Comment (cdrage) [inline extensions/kind/package.json]:
```suggestion
          "description": "HTTPS Port"
```

File: extensions/kind/scripts/build.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+const zipper = require('zip-local');
+const path = require('path');
+const package = require('../package.json');
+// const mkdirp = require('mkdirp');
```
Comment (cdrage) [inline extensions/kind/scripts/build.js]:
```suggestion
```

File: extensions/kind/scripts/build.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+zipper.sync.zip(path.resolve(__dirname, '../')).compress().save(destFile);
+
+//
+// create unzipped built-in
+/* Remove it to enable kind extension in the production assembly
+const unzippedDirectory = path.resolve(builtinDirectory, `${package.name}.cdix`);
+mkdirp(unzippedDirectory).then(() => {
+  zipper.sync.unzip(destFile).save(unzippedDirectory);
+});
+*/
```
Comment (cdrage) [inline extensions/kind/scripts/build.js]: remove commented code

File: extensions/kind/src/extension.ts [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    }
+
+    // grab http host port
+    let httpHostPort = 9090;
+    if (params['kind.cluster.creation.http.port']) {
+      httpHostPort = params['kind.cluster.creation.http.port'];
+    }
+
+    // grab https host port
+    let httpsHostPort = 9090;
```
Comment (cdrage) [inline extensions/kind/src/extension.ts]: shouldn't this be 9443?

**Review by cdrage — CHANGES_REQUESTED** [review]
Was able to test this out and it worked perfectly! Some little nitpicks, but was able to correctly use the cluster, make changes, etc.

One thing to note too, is that the terminal's text is right to the edge on the left, as well as it's hard to read with the red font. 

I'm assuming this is normal since we do not make any UI/UX changes in this PR and by default it's red?

**Comment by cdrage** [comment]
Screenshot 2022-12-20 at 3 59 18 PM

The red font as well as the text being right to the edge makes it difficult to read, perhaps we should change it to the default white font we use + the spacing.

**Comment by benoitf** [comment]
I think kind is logging to stderr and not stdout ( for the color)

File: extensions/kind/scripts/build.js (hunk 4 above)
Comment (benoitf) [reply extensions/kind/scripts/build.js]: I let it there as idea was to embed it by default for the next release, but removing the comment

**Comment by benoitf** [comment]
@cdrage comments fixed

**Review by cdrage — APPROVED** [review]
Tested and works well! Thanks for the log colouring changes.

The only nitpick I have is that the error is at the very bottom and not inline with our error / warnings output. But that's something we can fix with the UI/UX team when they refactor these sections.

LGTM!


Screenshot 2022-12-21 at 3 46 25 PM

Screenshot 2022-12-21 at 3 49 52 PM

**Pull request merged**